project/src/main/java/com/google/sps/Models/Database.java

```diff
@@ -128,59 +128,54 @@ public static Document createDocument(String name, String language, String hash,
       docEntity.setProperty("userIDs", userIDs);
       getDatastore().put(docEntity);
 
+      addDocumentForUser(hash, userID);
       return new Document(name, language, hash, userIDs);
   }
 
   public static Document getDocumentByHash(String hash) {
     Query query = new Query("Document").addFilter("hash", Query.FilterOperator.EQUAL, hash);
     Entity docEntity = getDatastore().prepare(query).asSingleEntity();
 
-    String name = (String) docEntity.getProperty("name");
-    String language = (String) docEntity.getProperty("language");
-    ArrayList<Long> userIDs = (ArrayList)docEntity.getProperty("userIDs");
-
     if(docEntity == null) {
       return null;
     }
 
+    String name = (String) docEntity.getProperty("name");
+    String language = (String) docEntity.getProperty("language");
+    ArrayList<Long> userIDs = (ArrayList)docEntity.getProperty("userIDs");
     return new Document(name, language, hash, userIDs);
   }
 
   public static ArrayList<Long> getDocumentUsers(String hash) {
     Query query = new Query("Document").addFilter("hash", Query.FilterOperator.EQUAL, hash);
     Entity docEntity = getDatastore().prepare(query).asSingleEntity();
 
-    ArrayList<Long> userIDs = (ArrayList)docEntity.getProperty("userIDs");
-
     if(docEntity == null) {
       return null;
     }
 
+    ArrayList<Long> userIDs = (ArrayList) docEntity.getProperty("userIDs");
```

Comment:
Is there a particular reason that you used getListProperty() to access a list property of an entity earlier and now you are using the standard getProperty()?

Reply:
getListProperty is a function i made that returns an empty arrayList if the list returned is null; this is because the Datastore automatically stores empty lists as null and these retrievals would otherwise throw NullPointerExceptions

Reply:
Right, so is there a reason that you are not using that here to avoid that issue?